src/analysis/mean_inequalities.lean

```diff
@@ -103,47 +108,206 @@ begin
+    rw [← div_le_iff hG, sum_div, ← rpow_le_rpow_iff hpq.pos, ← rpow_mul,
+      one_div_mul_cancel hpq.ne_zero, rpow_one],
+    -- Now the inequality follows from the weighted generalized mean inequality
+    -- with weights `w_i` and numbers `z_i` given by the following formulas.
+    set w : ι → ℝ≥0 := λ i, (g i) ^ q / G ^ q,
+    set z : ι → ℝ≥0 := λ i, f i * (G / g i) ^ (q / p),
+    -- Show that the sum of weights equals one
+    have A : ∑ i in s, w i = 1,
+    { rw [← sum_div, hGq, div_self],
+      simpa [rpow_eq_zero_iff, hpq.symm.ne_zero] using hG },
+    -- LHS of the goal equals LHS of the weighted generalized mean inequality
+    calc (∑ i in s, f i * g i / G) ^ p = (∑ i in s, w i * z i) ^ p :
+      begin
+        congr, ext i,
+        have : q - q / p = 1, by field_simp [hpq.ne_zero, hpq.symm.mul_eq_add],
+        dsimp only [w, z],
+        rw [← div_rpow, mul_left_comm, mul_div_assoc, ← @inv_div _ _ _ G, inv_rpow,
+          ← div_eq_mul_inv, ← rpow_sub']; simp [this]
+      end
+    -- Apply the generalized mean inequality
+    ... ≤ ∑ i in s, w i * (z i) ^ p : nnreal.rpow_am_le_am_rpow s w z A (le_of_lt hpq.one_lt)
+    -- Simplify the right hand side. Terms with `g i ≠ 0` are equal to `(f i) ^ p`,
+    -- the others are zeros.
+    ... ≤ ∑ i in s, (f i) ^ p :
+      begin
+        refine sum_le_sum (λ i hi, _),
+        dsimp only [w, z],
+        rw [mul_rpow, mul_left_comm, ← rpow_mul _ _ p, div_mul_cancel _ hpq.ne_zero, div_rpow,
+          div_mul_div, mul_comm (G ^ q), mul_div_mul_right],
+        nth_rewrite 1 [← mul_one ((f i) ^ p)],
+        exact canonically_ordered_semiring.mul_le_mul (le_refl _) (div_self_le _),
+        simpa [hpq.symm.ne_zero] using hG
+      end }
+end
+
+theorem is_greatest_Lp (f : ι → ℝ≥0) {p q : ℝ} (hpq : p.is_conjugate_exponent q) :
+  is_greatest ((λ g : ι → ℝ≥0, ∑ i in s, f i * g i) ''
+    {g | ∑ i in s, (g i)^q ≤ 1}) ((∑ i in s, (f i)^p) ^ (1 / p)) :=
+begin
+  split,
+  { use λ i, ((f i) ^ p / f i / (∑ i in s, (f i) ^ p) ^ (1 / q)),
+    by_cases hf : ∑ i in s, (f i)^p = 0,
+    { simp [hf, hpq.ne_zero, hpq.symm.ne_zero] },
+    { have A : p + q - q ≠ 0, by simp [hpq.ne_zero],
+      have B : ∀ y : ℝ≥0, y * y^p / y = y^p,
+      { refine λ y, mul_div_cancel_left_of_imp (λ h, _),
+        simpa [h, hpq.ne_zero] },
+      simp only [set.mem_set_of_eq, div_rpow, ← sum_div, ← rpow_mul,
+        div_mul_cancel _ hpq.symm.ne_zero, rpow_one, div_le_iff hf, one_mul, hpq.mul_eq_add,
+        ← rpow_sub' _ A, _root_.add_sub_cancel, le_refl, true_and, ← mul_div_assoc, B],
+      rw [div_eq_iff, ← rpow_add hf, hpq.inv_add_inv_conj, rpow_one],
+      simpa [hpq.symm.ne_zero] using hf } },
+  { rintros _ ⟨g, hg, rfl⟩,
+    apply le_trans (inner_le_Lp_mul_Lq s f g hpq),
+    simpa only [mul_one] using canonically_ordered_semiring.mul_le_mul (le_refl _)
+      (nnreal.rpow_le_one hg (le_of_lt hpq.symm.one_div_pos)) }
+end
+
+theorem minkowskii (f g : ι → ℝ≥0) {p : ℝ} (hp : 1 ≤ p) :
```

Comment:
There was a discussion in Zulip about the name of this theorem. We agreed that Minkowski was too short (there are many things due to Minkowski!) and we converged to `sum_rpow_minkowski` (if you want to put two `i`s, I 'll trust you on this!). As for Hölder, we came up with soem name `sum_rpow_holder`, but I think the name you used above is good also. You should absolutely put a docstring mentioning Hölder on this inequality, though, to make it easily findable with `grep`.